### ISSUE TITLE
Set fixed limit as default back pressure configuration

### ISSUE THREAD

**Issue by npepinpe** [camunda/zeebe#5113]
**Is your feature request related to a problem? Please describe.**

The current default back pressure configuration uses the Vegas algorithm with an initial limit of 20. What this means is it will poorly handle initial bursts, and may result in a poor first user experience - it's also generally quite pessimistic and more difficult to configure/explain to users.

See: 

- https://github.com/zeebe-io/zeebe/issues/4765
- https://github.com/zeebe-io/zeebe/issues/3367
- https://forum.zeebe.io/t/resource-exhausted-reached-maximum-capacity-of-requests-handled/731

**Describe the solution you'd like**

After discussing it with @deepthidevaki and @menski, the current proposal would be to switch to a fixed limit, with a default limit configured based on the Camunda Cloud Development Cluster (single node, single partition) machine for a relatively low maximum command latency target, say 1s, during a stress test.

**Describe alternatives you've considered**

Using AIMD as default - while more adaptive, the downside here is that AIMD, under duress, will lower the limit lower than it really needs to. However it's also quite easy to explain to users, so could be a viable alternative.

**Comment by Zelldon** [comment]
I played today a bit around with that I will summarize here my observations:


### Setup

I used one broker, one partition, on starter which starts 100 workflow instances per second and 6 of our workers with 120 job activation. I only changed between the benchmarks the Backpressure algorithms and limits.

Example configuration:

[zeebe-values.yaml.txt](https://github.com/zeebe-io/zeebe/files/5142717/zeebe-values.yaml.txt)

| Algorithm | Limit | Processing | Latency |
|--------------|--------|----------------|-------------|
| Default | 20 | ![zell-default-processing](https://user-images.githubusercontent.com/2758593/91581083-4a361280-e94e-11ea-8794-12a8da0aa00a.png) | ![zell-default-latency](https://user-images.githubusercontent.com/2758593/91581107-528e4d80-e94e-11ea-83ad-05c5562b6d6a.png) |
| Fixed | 10 | ![zell-fixed-10-single-part-100-workflows-processing](https://user-images.githubusercontent.com/2758593/91582264-d72d9b80-e94f-11ea-80bf-8ab127ac7636.png) | ![zell-fixed-10-single-part-100-workflows-latency](https://user-images.githubusercontent.com/2758593/91582262-d5fc6e80-e94f-11ea-94f5-89f86c866efc.png) |
| Fixed | 20 | ![zell-fixed-20-single-part-100-workflows-processing](https://user-images.githubusercontent.com/2758593/91581160-646ff080-e94e-11ea-9d5e-52072bfe1a7b.png) | ![zell-fixed-20-single-part-100-workflows-latency](https://user-images.githubusercontent.com/2758593/91581157-63d75a00-e94e-11ea-9948-abb3ad433131.png) |
| Fixed | 30 | ![zell-fixed-30-single-part-100-workflows-processing](https://user-images.githubusercontent.com/2758593/91582299-e7457b00-e94f-11ea-96b2-2deb81b8f29c.png) | ![zell-fixed-30-single-part-100-workflows-latency](https://user-images.githubusercontent.com/2758593/91582296-e6ace480-e94f-11ea-90a2-11ee8abdd4c3.png) |
| Fixed | 100 | ![zell-fixed-100-single-part-100-workflows-processing](https://user-images.githubusercontent.com/2758593/91581236-7ce00b00-e94e-11ea-84d1-503a579bc56c.png) | ![zell-fixed-100-single-part-100-workflows-latency](https://user-images.githubusercontent.com/2758593/91581233-7c477480-e94e-11ea-8b4e-6f96590cb972.png) |
| Fixed | 300 | ![zell-fixed-300-single-part-100-workflows-processing](https://user-images.githubusercontent.com/2758593/91581294-92553500-e94e-11ea-9201-6957b0ddce5f.png) | ![zell-fixed-300-single-part-100-workflows-latency](https://user-images.githubusercontent.com/2758593/91581292-91240800-e94e-11ea-960c-e4178f6b0e33.png) | 

Based on the results we can see that the limit doesn't affect the throughput, but it significatly changes the processing latency, which I think is expected. The bigger the limit the higher is also the latency, so it getting worse if you increase the limit. I think with a limit of 10 I got some quite good results.

**Comment by menski** [comment]
I did some testing with the `gradient` algorithm, and the configuration 
```
  - name: ZEEBE_BROKER_BACKPRESSURE_ALGORITHM
    value: "gradient"
  - name: ZEEBE_BROKER_BACKPRESSURE_GRADIENT_RTTTOLERANCE
    value: "1.0"
  - name: ZEEBE_BROKER_BACKPRESSURE_GRADIENT_MINLIMIT
    value: "10"
  - name: ZEEBE_BROKER_BACKPRESSURE_GRADIENT_INITIALLIMIT
    value: "100"
```

## Performance with 1 partition, 6 clients and 1 starter (100req/s)
![gradient-1partition-processing](https://user-images.githubusercontent.com/516296/92600774-4ab69d80-f2ac-11ea-806d-2a3f3ae56572.png)
![gradient-1partition-latency](https://user-images.githubusercontent.com/516296/92600771-4a1e0700-f2ac-11ea-8f66-763e66302b9d.png)


## Performance with 3 partitions, 12 clients and 1 starter (300req/s)
![gradient-3partitions-processing](https://user-images.githubusercontent.com/516296/92600780-4b4f3400-f2ac-11ea-9b9f-99fcb5318dc7.png)
![gradient-3partitions-latency](https://user-images.githubusercontent.com/516296/92600776-4b4f3400-f2ac-11ea-879b-177d7a9f1db2.png)

## Burst tests in the cloud (Development - XS)

###  Burst script

```
#!/bin/bash -u

JOBS=${1:-4}
INSTANCES=${2:-100}
PROCESS_ID=${3:-start-event}
JOBLOG=$(mktemp)

trap "{ rm -f $JOBLOG; }" EXIT

echo "Creating $INSTANCES instances, with concurrency of $JOBS"

parallel -n0 --joblog $JOBLOG -j$JOBS zbctl create instance $PROCESS_ID 2>/dev/null >/dev/null ::: $(seq 1 $INSTANCES)

awk 'FNR <= 1 {ok=0; error=0; next} $7 == 0 {ok+=1} $7 > 0 {error+=1} END {print "Total: " (ok+error); print "   Ok: " ok; print "Error: " error; print " Rate: " (ok * 100 / (ok + error)) "%"}' $JOBLOG
```

### Results

#### Default Backpressure (3 Runs)

```
Creating 250 instances, with concurrency of 10
Total: 250
   Ok: 247
Error: 3
 Rate: 98.8%
Creating 250 instances, with concurrency of 20
Total: 250
   Ok: 171
Error: 79
 Rate: 68.4%
Creating 250 instances, with concurrency of 25
Total: 250
   Ok: 170
Error: 80
 Rate: 68%
Creating 250 instances, with concurrency of 50
Total: 250
   Ok: 162
Error: 88
 Rate: 64.8%
Creating 250 instances, with concurrency of 75
Total: 250
   Ok: 161
Error: 89
 Rate: 64.4%
Creating 250 instances, with concurrency of 100
Total: 250
   Ok: 177
Error: 73
 Rate: 70.8%
```

```
Creating 250 instances, with concurrency of 10
Total: 250
   Ok: 241
Error: 9
 Rate: 96.4%
Creating 250 instances, with concurrency of 20
Total: 250
   Ok: 193
Error: 57
 Rate: 77.2%
Creating 250 instances, with concurrency of 25
Total: 250
   Ok: 196
Error: 54
 Rate: 78.4%
Creating 250 instances, with concurrency of 50
Total: 250
   Ok: 202
Error: 48
 Rate: 80.8%
Creating 250 instances, with concurrency of 75
Total: 250
   Ok: 206
Error: 44
 Rate: 82.4%
Creating 250 instances, with concurrency of 100
Total: 250
   Ok: 234
Error: 16
 Rate: 93.6%
```

```
Creating 250 instances, with concurrency of 10
Total: 250
   Ok: 244
Error: 6
 Rate: 97.6%
Creating 250 instances, with concurrency of 20
Total: 250
   Ok: 198
Error: 52
 Rate: 79.2%
Creating 250 instances, with concurrency of 25
Total: 250
   Ok: 209
Error: 41
 Rate: 83.6%
Creating 250 instances, with concurrency of 50
Total: 250
   Ok: 193
Error: 57
 Rate: 77.2%
Creating 250 instances, with concurrency of 75
Total: 250
   Ok: 223
Error: 27
 Rate: 89.2%
Creating 250 instances, with concurrency of 100
Total: 250
   Ok: 229
Error: 21
 Rate: 91.6%
```

#### Gradient Backpressure (3 Runs)

```
Creating 250 instances, with concurrency of 10
Total: 250
   Ok: 250
Error: 0
 Rate: 100%
Creating 250 instances, with concurrency of 20
Total: 250
   Ok: 250
Error: 0
 Rate: 100%
Creating 250 instances, with concurrency of 25
Total: 250
   Ok: 250
Error: 0
 Rate: 100%
Creating 250 instances, with concurrency of 50
Total: 250
   Ok: 250
Error: 0
 Rate: 100%
Creating 250 instances, with concurrency of 75
Total: 250
   Ok: 250
Error: 0
 Rate: 100%
Creating 250 instances, with concurrency of 100
Total: 250
   Ok: 208
Error: 42
 Rate: 83.2%
```

```
Creating 250 instances, with concurrency of 10
Total: 250
   Ok: 250
Error: 0
 Rate: 100%
Creating 250 instances, with concurrency of 20
Total: 250
   Ok: 250
Error: 0
 Rate: 100%
Creating 250 instances, with concurrency of 25
Total: 250
   Ok: 243
Error: 7
 Rate: 97.2%
Creating 250 instances, with concurrency of 50
Total: 250
   Ok: 209
Error: 41
 Rate: 83.6%
Creating 250 instances, with concurrency of 75
Total: 250
   Ok: 206
Error: 44
 Rate: 82.4%
Creating 250 instances, with concurrency of 100
Total: 250
   Ok: 206
Error: 44
 Rate: 82.4%
```

```
Creating 250 instances, with concurrency of 10
Total: 250
   Ok: 250
Error: 0
 Rate: 100%
Creating 250 instances, with concurrency of 20
Total: 250
   Ok: 227
Error: 23
 Rate: 90.8%
Creating 250 instances, with concurrency of 25
Total: 250
   Ok: 226
Error: 24
 Rate: 90.4%
Creating 250 instances, with concurrency of 50
Total: 250
   Ok: 239
Error: 11
 Rate: 95.6%
Creating 250 instances, with concurrency of 75
Total: 250
   Ok: 222
Error: 28
 Rate: 88.8%
Creating 250 instances, with concurrency of 100
Total: 250
   Ok: 242
Error: 8
 Rate: 96.8%
```

**Comment by aivinog1** [comment]
@Zelldon @npepinpe Hi! I would like to take this one. It is pretty clear to me :)

**Comment by npepinpe** [comment]
Downgrading for now and marking this as needs stakeholder input, primarily because it's unnecessary for Camunda Cloud and only affects users deploying Zeebe by themselves, so needs some PM input on the priority.

**Comment by gabvillacis** [comment]
In my use case it is common to have instantiation bursts. The fixed algorithm with a limit of 10 is not working well for me, I am testing on my microservice stack and I have too many requests for instantiation of workflows rejected (around 40%). For now I decided to increase the limit to 50 because with that configuration I find acceptable response times and I have managed to lower the rejection percentage to 30%, but ultimately it is not enough. I'm going to try the Gradient algorithm although I must admit that I have a bit of a hard time getting familiar with it and my date to go into production is approaching.
